src/workerd/io/io-context.h

```diff
@@ -1029,15 +1029,20 @@ jsg::Promise<IoContext::MaybeIoOwn<addIoOwn, T>> IoContext::awaitIoImpl(
                   maybeException = kj::mv(maybeException),
                   maybeAsyncContext = kj::mv(maybeAsyncContext)](Worker::Lock& lock) mutable {
         jsg::AsyncContextFrame::Scope asyncScope(lock, maybeAsyncContext);
-        KJ_IF_SOME(exception, maybeException) {
-          resolver.reject(lock, kj::mv(exception));
-        } else {
-          resolver.resolve(lock);
-        }
+        resolver.resolve(kj::mv(maybeException));
```

Comment:
A comment here explaining this would be helpful for folks coming into this later.